platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp

```diff
@@ -31,9 +32,12 @@ SC_AGENT_IMPLEMENTATION(GenerateResponseAgent)
       utils::IteratorUtils::getAnyByOutRelation(&m_memoryCtx, otherAddr, scAgentsCommon::CoreKeynodes::rrel_1);
   ScAddr const & answerAddr =
       utils::IteratorUtils::getAnyByOutRelation(&m_memoryCtx, otherAddr, scAgentsCommon::CoreKeynodes::rrel_2);
+  ScAddr const & conceptAddr =
+      utils::IteratorUtils::getAnyByOutRelation(&m_memoryCtx, messageAddr, Keynodes::rrel_entity);
 
   ScAddr messageAnswer;
   ScAddr const actionToApply = createActionNode(messageAddr);
+  m_memoryCtx.CreateEdge(ScType::EdgeAccessConstPosPerm, actionToApply, conceptAddr);
```

Comment:
It is not correct for all actions to apply. You need some common logic to attach arguments from kb <message, action>. As it seems to me, this logic is in `createActionNode` method